crates/proof-of-sql/src/sql/proof_gadgets/sign_expr.rs

```diff
@@ -159,7 +162,13 @@ fn verify_bit_decomposition<S: ScalarExt>(
             rhs += S::from_wrapping(mult) * bit_eval;
         }
     }
-    (rhs == expr_eval)
+    let bits_that_must_match_inverse_lead_bit = U256::MAX
+        .shl(num_bits_allowed.unwrap_or(S::MAX_BITS).min(S::MAX_BITS) - 1)
```

Comment:
NIT: The `min` is a bit concerning here. I'd rather make this return an error if the allowed bits is larger than MAX_BITS.